vueui/src/components/view/cooking/microwave.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <div>
+    <template v-if="on">
+      <p>{{ cookingMessage }}</p>
```

Comment:
I don't like use of computed property to affect message. `pastHalfTime` could be replaced with `$root.$data.wtime > (start_time + (cook_time / 2))`
```suggestion
      <template v-if="pastHalfTime">
        <p v-if="!ingredientsPresent">There's nothing inside. Microwave works, though.</p>
        <p v-else-if="failed">Something doesn't look right...</p>
        <p v-else>It's cooking nicely!</p>
      </template>
      <p v-else>It's starting to cook...</p>
```